File: Solutions/2191- Sort the Jumbled Numbers/Solution.java
import java.util.*;

class Solution {
    public int[] sortJumbled(int[] mapping, int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        List<Integer> list = new ArrayList<>();
        for (int num : nums) {
            int mappedValue = 0;
            int index = 1;
            int temp = num;
            do {
                int digit = temp % 10;
                mappedValue += mapping[digit] * index;
                temp /= 10;
                index *= 10;
            } while (temp > 0);
            map.put(num, mappedValue);
            list.add(num);
        }

        list.sort((a, b) -> map.get(a).compareTo(map.get(b)));

        int index = 0;
        for (int item : list) {
            nums[index] = item;
            index++;
        }

        return nums;



    }
}

File: Solutions/2191- Sort the Jumbled Numbers/Solution.py
class Solution:
    def sortJumbled(self, mapping: List[int], nums: List[int]) -> List[int]:
        my_dict = dict()
        for num in nums:
            mapped_value = 0
            index = 1
            temp = num
            while True:
                digit = temp % 10
                mapped_value += mapping[digit] * index
                temp //= 10
                index *= 10
                if temp == 0:
                    break
            my_dict[num] = mapped_value

        return sorted(nums, key= lambda n: my_dict[n])